src/client/client-patch-browser.ts

```diff
@@ -104,6 +104,13 @@ const patchDynamicImport = (base: string, orgScriptElm: HTMLScriptElement) => {
             type: 'application/javascript',
           })
         );
+
+        // Apply CSP nonce to the script tag if it exists
+        const nonce = plt.$nonce$ ?? (window as any).nonce;
```

Comment:
Passing thought - instead of pulling this value off `window`, could we instead pull it as an attribute off the `html` root element?

To me, it seems the usage pattern here is currently:
- `<script nonce="value">` - for a single script
- `window.nonce` - for everything in the DOM

Whereas it could be:
- `<html nonce="value">` - for everything in the DOM
- `<script nonce="value">` - for a single script

Consistent DX with less brittleness tied to the `window` object. 

Do we have access to the root element in the execution context here?

Reply:
The way that Rails handles this is by adding a meta tag in the header which subsequent scripts can read from before injecting their content:

`<meta name="csp-nonce" content="nonce_inserted_here">`

I think this is preferable to setting on the main html tag (though I agree the DX is better to do that than window).

Reply:
@sean-perkins @cscorley Thanks for the feedback and suggestions! I like the idea of using the `meta` tag. I'll check it out and see the feasibility to replace the `window` approach!